server/src/handle_chaos_message.rs

```diff
@@ -0,0 +1,482 @@
+        }
+    }
+}
+
+impl From<&chaos_proto::chaos::Cause> for Cause {
+    fn from(proto: &chaos_proto::chaos::Cause) -> Cause {
+        Cause {
+            id: proto.get_id().to_string(),
+            wording: proto.get_wording().to_string(),
+            category: proto.get_category().get_name().to_string(),
+            created_at: ts_to_dt(proto.get_created_at()),
+            updated_at: ts_to_dt(proto.get_created_at()),
+        }
+    }
+}
+
+impl From<&chaos_proto::gtfs_realtime::TimeRange> for DateTimePeriod {
+    fn from(proto: &chaos_proto::gtfs_realtime::TimeRange) -> DateTimePeriod {
+        DateTimePeriod {
+            start: NaiveDateTime::from_timestamp(proto.get_start() as i64, 0),
+            end: NaiveDateTime::from_timestamp(proto.get_end() as i64, 0),
+        }
+    }
+}
+
+impl From<&chaos_proto::chaos::Tag> for Tag {
+    fn from(proto: &chaos_proto::chaos::Tag) -> Tag {
+        Tag {
+            id: proto.get_id().to_string(),
+            name: proto.get_id().to_string(),
+            created_at: ts_to_dt(proto.get_created_at()),
+            updated_at: ts_to_dt(proto.get_created_at()),
+        }
+    }
+}
+
+impl From<&chaos_proto::chaos::Message> for Message {
+    fn from(proto: &chaos_proto::chaos::Message) -> Message {
+        let channel = proto.get_channel();
+        Message {
+            text: proto.get_text().to_string(),
+            channel_id: channel.get_id().to_string(),
+            channel_name: channel.get_name().to_string(),
+            channel_content_type: channel.get_content_type().to_string(),
+            channel_types: channel.get_types().iter().map(|t| t.into()).collect(),
+            created_at: ts_to_dt(proto.get_created_at()),
+            updated_at: ts_to_dt(proto.get_created_at()),
+        }
+    }
+}
+
+impl From<&chaos_proto::chaos::Channel_Type> for ChannelType {
+    fn from(proto: &chaos_proto::chaos::Channel_Type) -> ChannelType {
+        use chaos_proto::chaos::Channel_Type;
+        match proto {
+            Channel_Type::web => ChannelType::Web,
+            Channel_Type::sms => ChannelType::Sms,
+            Channel_Type::email => ChannelType::Email,
+            Channel_Type::mobile => ChannelType::Mobile,
+            Channel_Type::notification => ChannelType::Notification,
+            Channel_Type::twitter => ChannelType::Twitter,
+            Channel_Type::facebook => ChannelType::Facebook,
+            Channel_Type::unkown_type => ChannelType::UnknownType,
+            Channel_Type::title => ChannelType::Title,
+            Channel_Type::beacon => ChannelType::Beacon,
+        }
+    }
+}
+
+impl From<&chaos_proto::chaos::Pattern> for ApplicationPattern {
+    fn from(proto: &chaos_proto::chaos::Pattern) -> ApplicationPattern {
+        ApplicationPattern {
+            begin_date: NaiveDateTime::from_timestamp(proto.get_start_date() as i64, 0).date(),
+            end_date: NaiveDateTime::from_timestamp(proto.get_end_date() as i64, 0).date(),
```

Comment:
```suggestion
            begin_date: NaiveDateTime::from_timestamp(i64::from(proto.get_start_date()), 0).date(),
            end_date: NaiveDateTime::from_timestamp(i64::from(proto.get_end_date()) , 0).date(),
```
So we can easily see that the conversion will not panic

Reply:
Are you sure start_date() and end_date() are timestamps ?
It seems to be a date (number of days since something ?) https://github.com/CanalTP/chaos-proto/blob/6b2fea75cdb39c7850571b01888b550881027068/chaos.proto#L106-L108